packages/axelar-wasm-std/src/msg_id/mod.rs

```diff
@@ -0,0 +1,40 @@
+use std::{fmt::Display, str::FromStr};
+
+use cosmwasm_schema::cw_serde;
+
+use self::tx_hash_event_index::HexTxHashAndEventIndex;
+
+pub mod tx_hash_event_index;
+
+#[cw_serde]
+pub enum Error {
+    InvalidMessageID { id: String, expected_format: String },
+}
+
+/// Any message id format must implement this trait.
+/// The implementation must satisfy the following invariants:
+///
+/// * if m1 != m2 then from_str(m1) != from_str(m2) (two different strings cannot parse to the same message id)
+///
+/// * if t1 != t2 then to_string(t1) != to_string(t2) (two different message ids cannot serialize to the same string)
+///
+/// There should be only one string that can identify a given message.
+/// Take extra care to handle things like leading 0s, casing, etc.
+pub trait MessageId: FromStr + Display {}
```

Comment:
Where's this used or going to be used?

Reply:
the primary places this would be used is by the router (verifying the format matches) and by the verifier (parsing the string into proper structured data).